### DISCUSSION TITLE
Block web site

### DISCUSSION THREAD

**Discussion by PeterKnotek** [TechnitiumSoftware/DnsServer#926] [Q&A]
HI

One question
How block web site : aaaauto.any over web management in technitium DNS
- eg .cz or .sk or hu or more 
- aaaauto.* not work
- aaaauto.sk  work 

Can a regular expression be used??

Thanks

**Comment by ShreyasZare** [comment]
Thanks for asking. You just need to enter the domain name to in the Blocked section on the admin panel. This will block the domain name and all its subdomain names. There is no support for wildcard (the way you are trying) or regex directly but you can use the Advanced Blocking app and configure it to use regex. 

Note that you need to disable the default blocking feature in the DNS server when you are using the Advanced Blocking app else both blocking list will keep working together. The app needs to be configured by editing json text since there is no GUI support available yet. So, entering regex in json string will requiring escaping some characters so that the json format is valid.

**Reply by PeterKnotek** [reply]
Hi 
 },
  "networkGroupMap": {
   path to file .... with IP
   url to file .... ith IP 
  },

   it is possible ?

thanks

**Reply by ShreyasZare** [reply]
If you are referring to the Advanced Blocking app then no, the network group map config is to map a specific group to a client subnet. To use the app, you first need to define a group or edit the example group in the config and then map it to your subnet or map to "0.0.0.0/0" for mapping to all networks.

**Reply by PeterKnotek** [reply]
Olaaaa
How block domain name  in advanced blocking - for example google.xyz
any.google.any
For example :
www.google.sk or com or any  - block
www.xyz.google.xyz - block

Thanks

**Reply by ShreyasZare** [reply]
You can use `.\.google\..` as the regex for blocking `*.google.*` and `^google\..` for blocking `google.*`. Note that you will need to escape the `\` char while entering this regex in json config as string value.

You can refer to this regex documentation [here](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference).
